plugins/autoip.rb

```diff
@@ -35,29 +36,104 @@ def self.pre_validate options
       # Remove any 'DOMAIN\' prefix from the username
       username = options[:username]
       username = username.gsub(/^.+\\(.*)/, '\1')
-      puts "Requesting IP in #{options[:subnet]} vlan"
+      if (not options[:password])
+        print 'Password: '
+        options[:password] = STDIN.noecho(&:gets).chomp
+        puts ''
+      end
+
+
+      # Authenticate and get a token for further operations
+
+      puts "Requesting auth token for #{username}"
+      uri = options[:add_uri].gsub(/APIAPP/, {
+        'APIAPP' => options[:apiapp],
+      })
+      uri = uri 
+      uri = URI.escape(uri)
+      uri = URI.parse(uri.concat("user/"))
```

Comment:
when reversing these uri assignments I would get the error below.

/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/uri/common.rb:304:in `escape': undefined method `gsub' for #<URI::HTTPS:0x007fb272ccf728> (NoMethodError)
        from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/uri/common.rb:623:in `escape'
        from /Users/dmorris/src/mkvm/plugins/autoip.rb:54:in `pre_validate'
        from ./mkvm.rb:88:in `block in <main>'
        from ./mkvm.rb:88:in `each'
        from ./mkvm.rb:88:in `<main>'